p2p/pex/pex_reactor.go

```diff
@@ -543,6 +543,12 @@ func (r *Reactor) dialAttemptsInfo(addr *p2p.NetAddress) (attempts int, lastDial
 }
 
 func (r *Reactor) dialPeer(addr *p2p.NetAddress) error {
+	if r.Switch.Peers().Size() == 0 {
+		r.Logger.Error("Peer Info", "numPeers", r.Switch.Peers().Size())
+	} else {
+		r.Logger.Info("Peer Info", "numPeers", r.Switch.Peers().Size())
+	}
+
```

Comment:
Apologies for reopening the discussion, but I would like to understand the motivation behind logging here.

Here are some of my hypotheses. Please forgive my ignorance and fill me in with the correct information:

If you want to monitor the number of connected peers on the node, there are methods already in place for it: you can query the `/net_info` RPC endpoint, and I'm fairly sure there is also a Prometheus endpoint that will get you the information.

If you wanted to put this information in the log file specifically, why having no peers is an error message while having peers is an info message? Maybe emitting a debug message in every case is a more homogenous solution. Maybe there is an implicit error that zero peers bring that I'm unaware of.

Also, node operators tend to look at Prometheus and RPC for node monitoring before they start parsing the logs. I understand that developers usually start parsing the logs before turning to other troubleshooting methods.

Is this logging for developer troubleshooting purposes? If that's the case and the number of peers doesn't impact execution, I would still consider a debug message a better log level.

Finally, I'm trying to understand why the logging happens precisely here. (In dialPeer and at the beginning of it). dialPeer is called three times in the whole p2p library (not including tests):
ReceiveAddr - if we receive a seed node address, let's call it immediately
ensurePeers - if we can add more peers, let's add more peers (ensure, that there are peers we connected to)
crawlPeers - Find new peers on the network and add them

Are these log messages expected to be emitted only in these cases?

The reason I'm asking these questions is because I'd like to port this to CometBFT main, if it adds functionality for either the developer or the end user. But I need to understand the use case better.

Reply:
Hey @greg-szabo , no problems. I'd let @pratikspatil024 jump in because I might be wrong here but AFAIR, as you mentioned, we wanted to add some logging on the peer stats. I directly ported that in from maticnetwork/tendermint#32. If these numbers are trivial to get from an RPC endpoint then I guess we can remove the logs.

Reply:
Yes, as @Raneet10 mentioned, the purpose of these logs was just to get the number of peers a node is connected to. We can remove these logs as we can get the information through RPC.